behaviour/graql/reasoner/resolution/type-hierarchy.feature

```diff
@@ -0,0 +1,583 @@
+    Given for each session, graql define
+      """
+      define
+
+      person sub entity,
+          plays writer,
+          plays performer,
+          plays film-writer,
+          plays actor,
+          plays scifi-writer,
+          plays scifi-actor;
+
+      performance sub relation,
+          relates writer,
+          relates performer;
+
+      film-production sub relation,
+          relates film-writer,
+          relates actor;
+
+      scifi-production sub film-production,
+          relates scifi-writer as film-writer,
+          relates scifi-actor as actor;
+
+      performance-to-scifi sub rule,
+      when {
+          (writer:$x, performer:$y) isa performance;
+      },
+      then {
+          (scifi-writer:$x, scifi-actor:$y) isa scifi-production;
+      };
+      """
+    Given for each session, graql insert
+      """
+      insert
+      $x isa person;
+      $y isa person;
+      (writer:$x, performer:$y) isa performance;
+      """
+    When materialised keyspace is completed
+    # super-roles, sub-relation
+    Then for graql query
+      """
+      match (film-writer:$x, actor:$y) isa scifi-production; get;
+      """
+    Then all answers are correct in reasoned keyspace
+    Then answer size in reasoned keyspace is: 1
+    Then materialised and reasoned keyspaces are the same size
+
+
+  Scenario: when a sub-relation is inferred, it can be retrieved by matching its super-relation and super-roles
+    Given for each session, graql define
+      """
+      define
+
+      person sub entity,
+          plays writer,
+          plays performer,
+          plays film-writer,
+          plays actor,
+          plays scifi-writer,
+          plays scifi-actor;
+
+      performance sub relation,
+          relates writer,
+          relates performer;
+
+      film-production sub relation,
+          relates film-writer,
+          relates actor;
+
+      scifi-production sub film-production,
+          relates scifi-writer as film-writer,
+          relates scifi-actor as actor;
+
+      performance-to-scifi sub rule,
+      when {
+          (writer:$x, performer:$y) isa performance;
+      },
+      then {
+          (scifi-writer:$x, scifi-actor:$y) isa scifi-production;
+      };
+      """
+    Given for each session, graql insert
+      """
+      insert
+      $x isa person;
+      $y isa person;
+      (writer:$x, performer:$y) isa performance;
+      """
+    When materialised keyspace is completed
+    # super-roles, super-relation
+    Then for graql query
+      """
+      match (film-writer:$x, actor:$y) isa film-production; get;
+      """
+    Then all answers are correct in reasoned keyspace
+    Then answer size in reasoned keyspace is: 1
+    Then materialised and reasoned keyspaces are the same size
+
+
+  Scenario: subtypes trigger rules based on their parents; parent types don't trigger rules based on their children
```

Comment:
make it clear that this to do with instances of data that exist, not to do with the query and the `then` of the rule that are executed.

Reply:
Updated to: ```
subtype instances trigger rules whose `when` block references their parent type; supertype instances do not```